### PR TITLE
Add insert_image_chip

### PR DESCRIPTION
**Pull request by arrufat**
I think this function would be very convenient in the Python bindings to be able to revert the `extract_image_chip` without exposing the `transform` types in Python.

Exposing the `transform` tooling could be done at a latter time if there's a need for that, but I guess this function should cover most of the basic cases, such as #2595.

### CODE REVIEW

**Comment by arrufat** [comment]
If you think it's needed, I could add a C++ version, but it's so easy to implement, that I don't think it's needed. The main reason why I added it to Python only, instead of exposing the `transform` type, was to avoid looping in Python.

**Comment by arrufat** [comment]
Hm, that's not right, I need to perform some kind of interpolation; otherwise I don't fill the full area... I will think more about it.

File: dlib/image_transforms/interpolation.h [hunk 1]
```diff
@@ -2001,6 +2001,50 @@ namespace dlib
...
+        const_image_view<image_type2> vchip(chip);
+        DLIB_CASSERT(static_cast<unsigned long>(vchip.nr()) == location.rows &&
+                     static_cast<unsigned long>(vchip.nc()) == location.cols,
+                     "The chip and the location do not have the same size.")
+        const auto tf = get_mapping_to_chip(location);
+        for (long r = 0; r < vimg.nr(); ++r)
+        {
+            for (long c = 0; c < vimg.nc(); ++c)
+            {
+                interp(vchip, tf(dlib::vector<double, 2>(c, r)), vimg[r][c]);
```
Comment (davisking) [inline dlib/image_transforms/interpolation.h:2028]:
```suggestion
                interp(vchip, tf(dpoint(c, r)), vimg[r][c]);
```
There is a `dpoint` alias for this (i.e. double point)

**Comment by davisking** [comment]
> Hm, that's not right, I need to perform some kind of interpolation; otherwise I don't fill the full area... I will think more about it.

Looks reasonable.  You happy with it or still trying stuff?

**Comment by arrufat** [comment]
Ah, I'm happy with it now :)

**Comment by davisking** [comment]
> Ah, I'm happy with it now :)

Sweet.  Thanks for the PR :)

**Pull request merged**